### PR TITLE
Fix FileWatcher and JsonRPC teardown

### PR DESCRIPTION
**Pull request by bschnurr**
- also change `JsonRpcWrapper` to an extension class so that we dont have multiple references to the the `JsonRPC` instance

fixes: https://github.com/microsoft/PTVS/issues/7651

### CODE REVIEW

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/FileWatcher/Listener.cs [hunk 1]
```diff
@@ -29,19 +27,16 @@
 namespace Microsoft.PythonTools.LanguageServerClient.FileWatcher {
     class Listener : IDisposable {
         private JsonRpc _rpc;
-        private JsonRpcWrapper _rpcWrapper;
```
Comment (bschnurr) [inline Python/Product/PythonTools/PythonTools/LanguageServerClient/FileWatcher/Listener.cs:32]: removed `JsonRpcWrapper` member

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/FileWatcher/Listener.cs [hunk 2]
```diff
@@ -29,19 +27,16 @@
...
+        private readonly IFileWatcherService _WorkspaceFileWatcher;
```
Comment (bschnurr) [inline Python/Product/PythonTools/PythonTools/LanguageServerClient/FileWatcher/Listener.cs:30]: added `_WorkspaceFileWatcher` to help unregister `OnFileChanged`

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/FileWatcher/Listener.cs [hunk 3]
```diff
@@ -118,7 +115,7 @@ class Listener : IDisposable {
         private async Task OnFileChanged(object sender, System.IO.FileSystemEventArgs e) {
 
             // Skip directory change events
```
Comment (bschnurr) [inline Python/Product/PythonTools/PythonTools/LanguageServerClient/FileWatcher/Listener.cs:117]:
```suggestion
            // Skip directory change events
            // Ignore filechanges events after _rpc has been disposed
```

File: Python/Product/Common/Infrastructure/JsonRpcWrapper.cs [hunk 4]
```diff
@@ -21,39 +21,30 @@
 using StreamJsonRpc;
 
 namespace Microsoft.PythonTools.Common.Infrastructure {
-    internal readonly struct JsonRpcWrapper {
+    internal static class JsonRpcExtensions {
```
Comment (bschnurr) [inline Python/Product/Common/Infrastructure/JsonRpcWrapper.cs:24]: main idea of the fix, make this static

**Review by StellaHuang95 — APPROVED** [review]

**Pull request merged**